x-pack/qa/password-protected-keystore/src/javaRestTest/java/org/elasticsearch/password_protected_keystore/ReloadSecureSettingsWithPasswordProtectedKeystoreRestIT.java

```diff
@@ -23,17 +27,42 @@
 import static org.hamcrest.Matchers.nullValue;
 
 public class ReloadSecureSettingsWithPasswordProtectedKeystoreRestIT extends ESRestTestCase {
-    // From build.gradle
-    private final String KEYSTORE_PASSWORD = "keystore-password";
-    private final int NUM_NODES = 2;
+
+    private static final String KEYSTORE_PASSWORD = "keystore-password";
+    private static final int NUM_NODES = 2;
+
+    @ClassRule
+    public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
+        .distribution(DistributionType.DEFAULT)
+        .nodes(NUM_NODES)
+        .keystorePassword(KEYSTORE_PASSWORD)
+        .setting("xpack.security.enabled", "true")
+        .setting("xpack.security.authc.anonymous.roles", "anonymous")
+        .setting("xpack.security.transport.ssl.enabled", "true")
+        .setting("xpack.security.transport.ssl.certificate", "transport.crt")
+        .setting("xpack.security.transport.ssl.key", "transport.key")
+        .setting("xpack.security.transport.ssl.key_passphrase", "transport-password")
+        .setting("xpack.security.transport.ssl.certificate_authorities", "ca.crt")
+        .rolesFile(Resource.fromClasspath("roles.yml"))
+        .configFile("transport.key", Resource.fromClasspath("ssl/transport.key"))
+        .configFile("transport.crt", Resource.fromClasspath("ssl/transport.crt"))
+        .configFile("ca.crt", Resource.fromClasspath("ssl/ca.crt"))
+        .user("admin_user", "admin-password")
+        .user("test-user", "test-user-password", "user_role", false)
+        .build();
+
+    @Override
+    protected String getTestRestCluster() {
+        return cluster.getHttpAddresses();
+    }
 
     @SuppressWarnings("unchecked")
     public void testReloadSecureSettingsWithCorrectPassword() throws Exception {
         final Request request = new Request("POST", "_nodes/reload_secure_settings");
         request.setJsonEntity("{\"secure_settings_password\":\"" + KEYSTORE_PASSWORD + "\"}");
         final Response response = client().performRequest(request);
         final Map<String, Object> map = entityAsMap(response);
-        assertThat(ObjectPath.eval("cluster_name", map), equalTo("javaRestTest"));
+        assertThat(ObjectPath.eval("cluster_name", map), equalTo("test-cluster"));
```

Comment:
FWIW, you can name the cluster whatever you want with `.name("javaRestTest")` in cases where you don't want to go back and update a ton of existing tests.

Reply:
Went back and named the cluster as suggested 1091deeeb4e84720dc5f8581f7e092417c408b0d